src/tests/JIT/HardwareIntrinsics/Arm/Shared/Helpers.cs

```diff
@@ -2415,6 +2476,76 @@ private static long Reduce(Func<long, short, long> reduceOp, short[] op1)
 
         public static long AddPairwiseWideningAndAdd(long[] op1, int[] op2, int i) => (long)(op1[i] + AddWidening(op2[2 * i], op2[2 * i + 1]));
 
+        public static ulong AddCarryWideningEven(ulong[] op1, ulong[] op2, ulong[] op3, int i)
+        {
+            ulong lsb;
+            ulong res;
+
+            if (i < 0 || i >= op1.Length || i >= op2.Length || i >= op3.Length)
+                throw new ArgumentOutOfRangeException(nameof(i), "Index i is out of range");
+
+            if (i % 2 == 0)
+            {
+                if (i + 1 >= op3.Length)
```

Comment:
please add curley braces even if there is single statement inside if-conditions.

Also, there also needs validation for all the arrays that are accessed using the index `i`. e.g. , here, we should also check for `op2.Length`. Likewise, in other examples below.